examples/xhr-proxy/xhrproxy.js

```diff
@@ -0,0 +1,55 @@
+/* This is an example how to monkey-patch XMLHttpRequest() for HTTP requests to a local service port.
+ * Cockpit pages cannot use the standard XMLHttpRequest() due to its strict Content-Security-Policy,
+ * so these need to be wrapped into a raw cockpit channel and be "really" done from the host where
+ * the session runs.
+ *
+ * Note that non-localhost connections are still forbidden by this page's Content-Security-Policy
+ * (which can be changed in the manifest), and browsers blocking Cross-Origin requests.
+ *
+ * Documentation links:
+ * - https://cockpit-project.org/guide/latest/cockpit-channels.html
+ * - doc/protocol.md
+ * - doc/urls.md
+ */
+
+const origXHROpen = XMLHttpRequest.prototype.open;
+XMLHttpRequest.prototype.open = function(method, url) {
+    console.log(`calling patched XMLHttpRequest.open("${method}", "${url}")`);
+    const u = new URL(url);
+    if (u.protocol !== 'http:')
+        throw new Error("This demo only supports http:// URLs");
+
+    const channel = {
+        payload: "http-stream2",
+        method,
+        address: u.hostname,
+        port: u.port ? parseInt(u.port) : 80,
+        path: u.pathname + u.hash,
+    };
+
+    const channel_url = "/cockpit/channel/" + cockpit.transport.csrf_token + "?" +
+                        window.btoa(JSON.stringify(channel));
+
+    origXHROpen.apply(this, ["GET", channel_url]);
+}
+
```

Comment:
I'd suggest delineating which part is reusable (above), and which part is specific to the example ... below. Or perhaps putting the two sections in separate files.